Makefile.include

```diff
@@ -4,23 +4,43 @@
 all:
 
 # set undefined variables
-RIOTBASE ?= $(shell dirname "$(lastword $(MAKEFILE_LIST))")
-RIOTBASE := $(abspath $(RIOTBASE))
-
-CCACHE_BASEDIR := $(RIOTBASE)
-
-RIOTCPU ?= $(RIOTBASE)/cpu
-RIOTCPU := $(abspath $(RIOTCPU))
-
-RIOTBOARD ?= $(RIOTBASE)/boards
-RIOTBOARD := $(abspath $(RIOTBOARD))
-
-RIOTPKG ?= $(RIOTBASE)/pkg
+RIOTBASE       ?= $(shell dirname "$(lastword $(MAKEFILE_LIST))")
+CCACHE_BASEDIR ?= $(RIOTBASE)
+RIOTCPU        ?= $(RIOTBASE)/cpu
+RIOTBOARD      ?= $(RIOTBASE)/boards
+RIOTPKG        ?= $(RIOTBASE)/pkg
+RIOTPROJECT    ?= $(shell git rev-parse --show-toplevel 2>/dev/null || pwd)
```

Comment:
IIRC the RIOTPROJECT variable is meant to point to the project root of the
current application. Try it when building stuff in the RIOT-OS/applications
repo.
On Apr 5, 2016 2:51 PM, "Cenk Gündoğan" <notifications@github.com> wrote:

In Makefile.include
<https://github.com/RIOT-OS/RIOT/pull/5221#discussion_r58531659>:

> -
> -CCACHE_BASEDIR := $(RIOTBASE)
> -
> -RIOTCPU ?= $(RIOTBASE)/cpu
> -RIOTCPU := $(abspath $(RIOTCPU))
> -
> -RIOTBOARD ?= $(RIOTBASE)/boards
> -RIOTBOARD := $(abspath $(RIOTBOARD))
> -
> -RIOTPKG ?= $(RIOTBASE)/pkg
> +RIOTBASE       ?= $(shell dirname "$(lastword $(MAKEFILE_LIST))")
> +CCACHE_BASEDIR ?= $(RIOTBASE)
> +RIOTCPU        ?= $(RIOTBASE)/cpu
> +RIOTBOARD      ?= $(RIOTBASE)/boards
> +RIOTPKG        ?= $(RIOTBASE)/pkg
> +RIOTPROJECT    ?= $(shell git rev-parse --show-toplevel 2>/dev/null || pwd)

I may confuse something, but isn't RIOTPROJECT the same as RIOTBASE? Is
there a case where RIOTBASE does not point to the RIOT root?

—
You are receiving this because you were mentioned.

Reply to this email directly or view it on GitHub
<https://github.com/RIOT-OS/RIOT/pull/5221/files/cffc132d9f2fadf96ed546be911f8c70880187cc#r58531659>